sql/src/test/java/org/apache/druid/sql/calcite/util/SpecificSegmentsQuerySegmentWalker.java

```diff
@@ -388,13 +410,33 @@ public SegmentDescriptor getDescriptor()
           toolChest
       );
 
+
       // Wrap baseRunner in a runner that rewrites the QuerySegmentSpec to mention the specific segments.
       // This mimics what CachingClusteredClient on the Broker does, and is required for certain queries (like Scan)
       // to function properly.
-      return (theQuery, responseContext) -> baseRunner.run(
-          theQuery.withQuery(Queries.withSpecificSegments(theQuery.getQuery(), ImmutableList.copyOf(specs))),
-          responseContext
-      );
+      return (theQuery, responseContext) -> {
+        if (scheduler != null) {
+          Set<SegmentServerSelector> segments = new HashSet<>();
+          specs.forEach(spec -> segments.add(new SegmentServerSelector(null, spec)));
+          return scheduler.run(
+              scheduler.laneQuery(theQuery, segments),
+              new LazySequence<>(
+                  () -> baseRunner.run(
+                      theQuery.withQuery(Queries.withSpecificSegments(
+                          theQuery.getQuery(),
+                          ImmutableList.copyOf(specs)
+                      )),
+                      responseContext
+                  )
+              )
+          );
```

Comment:
Which tests cover this block?

Reply:
If you mean which tests use this block, then it is the [SqlResourceTest](https://github.com/apache/druid/pull/9407/files#diff-5a3523f6924e529221fa60db2bb56f05R125) which create this thing with a scheduler.

No laning strategy is currently using the set of `SegmentServerSelector` to make decisions at this time, I just went ahead and wired it up because we do have the set of `SegmentDescriptors` available here, in anticipation of some future usage. With the `ServerSelector` part of that set to `null` though, we would probably want to either have a mock or artifical set of servers to pretend to serve the test segments, so it might still need some future work?